repair/repair.cc

```diff
@@ -1346,6 +1345,32 @@ static int do_repair_start(seastar::sharded<database>& db, sstring keyspace,
         throw std::runtime_error("Node is not in NORMAL status yet!");
     }
 
+    std::vector<sstring> cfs;
+    std::vector<schema_ptr> schemas;
+    if (options.column_families.size()) {
+        cfs = options.column_families;
+        for (auto& cf : cfs) {
+            try {
+                schemas.push_back(db.local().find_column_family(keyspace, cf).schema());
+            } catch(...) {
+                throw std::runtime_error(format("No column family '{}' in keyspace '{}'", cf, keyspace));
+            }
+        }
+    } else {
+        std::tie(cfs, schemas) = list_column_families(db.local(), keyspace);
+    }
+    if (cfs.empty()) return id;
+    for (size_t idx = 1; idx < schemas.size(); ++idx) {
+        if (schemas[idx - 1]->get_partitioner() != schemas[idx]->get_partitioner()) {
+            throw std::runtime_error(
+                    format("All column families repaired together have to have the same partitioner. Different partitioners found: {} and {}",
+                        schemas[idx - 1]->get_partitioner(), schemas[idx]->get_partitioner()));
+        }
+    }
+    dht::i_partitioner& partitioner = schemas[0]->get_partitioner();
+
+    options.set_ranges(options_map, partitioner);
```

Comment:
The partitioner is not used inside set_ranges. Why do you pass it?

Reply:
This change is gone once I moved the check to repair_info constructor.